addons/purchase_stock/wizard/product_replenish.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class ProductReplenish(models.TransientModel):
+    _inherit = 'product.replenish'
+
+    supplier_id = fields.Many2one("product.supplierinfo", string="Vendor")
+
+    @api.model
+    def default_get(self, fields):
+        res = super(ProductReplenish, self).default_get(fields)
+        product_id = self.env['product.product'].browse(res.get('product_id'))
+        product_tmpl_id = product_id.product_tmpl_id
+        orderpoint = self.env['stock.warehouse.orderpoint'].search([('product_id', '=', product_tmpl_id.product_variant_id.id)])
+        if not orderpoint:
+            if product_tmpl_id.seller_ids:
+                orderpoint = self.env["stock.warehouse.orderpoint"].create({
+                    "product_id": product_tmpl_id.product_variant_id.id,
+                    "warehouse_id": res.get('warehouse_id', False),
+                    "supplier_id": product_tmpl_id.seller_ids[0].id
+                    })
+        res['supplier_id'] = orderpoint.supplier_id.id
```

Comment:
Is it necessary to create a new orderpoint ? If not, you can get back the supplier id directly:
```suggestion
        if orderpoint:
            res['supplier_id'] = orderpoint.supplier_id.id
        elif product_tmpl_id.seller_ids:
            res['supplier_id'] = product_tmpl_id.seller_ids[0].id
```